packages/foam-vscode/src/features/preview-navigation.spec.ts

```diff
@@ -50,3 +51,55 @@ describe('Stylable tag generation in preview', () => {
     );
   });
 });
+
+describe('Displaying included notes in preview', () => {
+  const noteA = createTestNote({
+    uri: 'note-a.md',
+    text: 'This is the text of note A',
+  });
+  const noteC = createTestNote({
+    uri: 'note-c.md',
+    text: 'This is the text of note C which includes ![[note-d]]',
+  });
+  const noteD = createTestNote({
+    uri: 'note-d.md',
+    text: 'This is the text of note D which includes ![[note-c]]',
+  });
+  const ws = new FoamWorkspace()
+    .set(noteA)
+    .set(noteC)
+    .set(noteD);
+  const md = markdownItWithNoteInclusion(MarkdownIt(), ws);
+
+  it('renders an included note', () => {
+    expect(
+      md.render(`This is the root node. 
+
+ ![[note-a]]`)
+    ).toMatch(
+      `<p>This is the root node.</p>
+<p><p>This is the text of note A</p>
+</p>`
+    );
+  });
+
+  it('displays the syntax when a note is not found', () => {
+    expect(
+      md.render(`This is the root node.
+    ![[note-b]]`)
+    ).toMatch(
+      `<p>This is the root node.
+![[note-b]]</p>
+`
+    );
+  });
+
+  it('displays a warning in case of cyclical inclusions', () => {
+    expect(md.render(noteD.source.text)).toMatch(
+      `<p>This is the text of note D which includes <p>This is the text of note C which includes <p>This is the text of note D which includes <div class=\"foam-cyclic-link-warning\">Cyclic link detected for wikilink: note-c</div></p>
```

Comment:
let's make the linter happy :)

Reply:
also, shouldn't the cycle be detected when `note C` imports `note D`?

Reply:
Interesting point. I now chose the same behaviour as vscode-memo. We could change it so that a note won't be included if it contains a reference to the same page. Not sure what the best option is in terms of user experience. Personally, I like it to easily see where a cyclic reference is made, instead of looking into the file. But, I do not hold a strong opinion on this.

Reply:
mmm.. maybe I wasn't clear or didn't understand the reply, what I meant is that right the cycle detection will include the source note.

`renders Note D -> renders Note C -> renders Note D -> cycle warning`

But the cycle actually happens a step ahead, so I was expecting something like:

`renders Note D -> renders Note C -> cycle warning`

does that makes sense? wdyt?